src/Picker/Picker.ios.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import * as React from 'react';
+import { Picker as RNPicker, View } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { StyleSheet } from '../PlatformStyleSheet';
+
+import type { Props } from './PickerTypes';
+
+export type State = {|
+  open: boolean,
+  selectedValue: string,
+  pickerValue: string,
+|};
+
+export default class Picker extends React.Component<Props, State> {
+  static defaultProps = {
+    optionsData: [],
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    const { selectedValue } = this.props;
+
+    this.state = {
+      open: false,
+      selectedValue,
+      pickerValue: selectedValue,
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const { selectedValue: newSelectedValue } = this.props;
+    const { selectedValue } = this.state;
+    if (
+      newSelectedValue !== prevProps.selectedValue &&
+      newSelectedValue !== selectedValue
+    ) {
+      this.setState({
+        selectedValue: newSelectedValue,
+        pickerValue: newSelectedValue,
+      });
+    }
+  };
+
+  handleOpenPress = () => {
+    this.setState(state => ({
+      open: !state.open,
+    }));
+  };
+
+  handlePickerValueChange = (value: string) => {
+    this.setState({ pickerValue: value });
+  };
+
+  handleOkPress = () => {
+    const { pickerValue } = this.state;
+    const { onValueChange } = this.props;
+    this.setState({
+      open: false,
+      selectedValue: pickerValue,
+    });
+    onValueChange(pickerValue);
+  };
+
+  render() {
+    const { open, selectedValue, pickerValue } = this.state;
+    const { optionsData } = this.props;
+    const selectedOption = optionsData.find(
```

Comment:
This logic, and the corresponding component is duplicated between ios and android. Could you extract it?

Reply:
Whole duplicate native picker implementation extracted to separate component and duplicate function added to `PickerHelpers`.